src/main/java/io/stargate/sgv2/jsonapi/service/operation/InsertOperationPage.java

```diff
@@ -4,29 +4,32 @@
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
 import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.exception.APIException;
+import io.stargate.sgv2.jsonapi.exception.APIExceptionCommandErrorBuilder;
 import io.stargate.sgv2.jsonapi.exception.mappers.ThrowableToErrorMapper;
 import io.stargate.sgv2.jsonapi.service.shredding.DocRowIdentifer;
 import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
 import java.util.function.Supplier;
 
 /**
- * The internal to insert operation results, keeping ids of successfully and not-successfully
- * inserted documents.
+ * Builds the response for an insert operation or one or more {@link InsertAttempt}s.
  *
- * <p>Can serve as an aggregator, using the {@link #aggregate} function. TODO: AARON DOCS
+ * <p>Keeps track of the inserts, their success status, and then builds the {@link CommandResult}
+ * via {@link #get()}.
  *
- * @param allInsertions Attempted insertions
- * @param returnDocumentResponses Whether to return detailed document responses
- * @param successfulInsertions Successfully inserted documents, NOTE: this list is mutated after
- *     creation
- * @param failedInsertions Failed insertions NOTE: this list is mutated after creation
+ * <p>Create an instance with all the insert attempts the insert operation will process, then call
+ * {@link #registerCompletedAttempt} for each completed attempt, the instance will then track failed
+ * and successful attempts. This is used as an aggregator for {@link
+ * io.smallrye.mutiny.groups.MultiCollect#in(Supplier, BiConsumer)}
  */
 public class InsertOperationPage implements Supplier<CommandResult> {
 
-  // TODO: AARON changed from a record because the successfulInsertions and failedInsertions were
-  // setable in the ctor
-  // but they have to be empty mutable lists
+  // All the insertions that are going to be attempted
   private final List<? extends InsertAttempt> allInsertions;
```

Comment:
adding comments to say this.

Reply:
Ok, sounds good.

Reply:
Changed the variable to be <InsertAttemp> and accept <? insertAttempt> in the ctor and up cast. This makes it easier for the Operations that use this because they wll have InsertAttempt subclasses.